### ISSUE TITLE
get_facebook_user() outputs a NULL result

### ISSUE THREAD

**Issue by alketii** [FrogSquare/GodotFireBase#2]
Hi,
I can successfully auth with Facebook,  but I am getting a NULL output from get_facebook_user().
This is my gdscript code
```
extends Node2D

onready var firebase = Globals.get_singleton("FireBase");
onready var sqlbridge = Globals.get_singleton("SQLBridge");

func _ready():
	firebase.initWithFile("res://godot-firebase-config.json",get_instance_ID())
	firebase.authConfig("'Google':false,'Facebook':true")


func _recive_message(from, key, data):
	if from == "FireBase":
		if key == "GoogleLogin" && data == "true": print("User Signed in.");
		if key == "FacebookLogin" && data == "true": print("User Signed in.");

func _on_facebook_login_pressed():
	firebase.facebook_sign_in()
	get_node("facebook_login").set_disabled(true)


func _on_get_profile_pressed():
	print(firebase.get_facebook_user())
```

And this is ADB logcat
```
--------- beginning of main
05-20 12:26:26.325 20573 20573 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-20 12:26:26.331 20573 20573 D AndroidRuntime: CheckJNI is OFF
05-20 12:26:26.501 20573 20573 D AndroidRuntime: Calling main entry com.android.commands.am.Am
05-20 12:26:26.541 20573 20573 D AndroidRuntime: Shutting down VM
05-20 12:26:26.903 20665 20665 I godot   : **INIT EVENT! - 0xb42a3230
05-20 12:26:26.903 20665 20665 I godot   : ***************** HELLO FROM JNI!!!!!!!!
05-20 12:26:26.903 20665 20665 I godot   : *******CLASS FOUND!!!
05-20 12:26:26.903 20665 20665 I godot   : STEP2, 0x10074a
05-20 12:26:26.903 20665 20665 I godot   : STEP3 -1229056644
05-20 12:26:26.903 20665 20665 I godot   : STEP4, 0x1d
05-20 12:26:26.903 20665 20665 I godot   : STEP4.5, 0x10074e
05-20 12:26:26.903 20665 20665 I godot   : STEP7
05-20 12:26:26.903 20665 20665 I godot   : STEP8
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _dir_open ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _dir_next ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _dir_close ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _dir_is_dir ok!!
05-20 12:26:26.903 20665 20665 I godot   : starting to attempt get methods
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _init_audio ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _write_buffer ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _quit ok!!
05-20 12:26:26.903 20665 20665 I godot   : *******GOT METHOD _pause ok!!
05-20 12:26:26.903 20665 20665 I godot   : CMDLINE LEN 0 - APK EXPANSION I
05-20 12:26:26.904 20665 20665 I godot   : ERROR: create: Condition ' !create_func ' is true. returned: 0
05-20 12:26:26.904 20665 20665 I godot   :    At: core/os/mutex.cpp:38.
05-20 12:26:26.904 20665 20665 I godot   : test construction 5
05-20 12:26:26.904 20665 20665 I godot   : running from dir /
05-20 12:26:26.904 20665 20665 I godot   : **SETUP
05-20 12:26:26.924 20665 20665 I godot   : has res dir: 
05-20 12:26:26.924 20665 20665 I godot   : exists engine cfg? 0
05-20 12:26:26.925 20665 20665 I godot   : loaded engine.cfg
05-20 12:26:26.942 20665 20665 I godot   : *****SETUP OK
05-20 12:26:26.973 20665 20665 I godot   : **START
05-20 12:26:27.168 20665 20695 I godot   : ^_^_^_^_^ newcontext -1458374368
05-20 12:26:27.168 20665 20695 I godot   : ^_^_^_^_^ resize -1458374368, 1920, 1080
05-20 12:26:27.169 20665 20695 I godot   : ANDROID MODULES : org/godotengine/godot/SQLBridge,org/godotengine/godot/FireBase
05-20 12:26:27.169 20665 20695 I godot   : mod count: 2
05-20 12:26:27.169 20665 20695 I godot   : LOADING MODULE: org/godotengine/godot/SQLBridge
05-20 12:26:27.170 20665 20695 I godot   : ****^*^*?^*^*class data 4d
05-20 12:26:27.172 20665 20695 I godot   : Singl:  SQLBridge Method: getBool RetVal: boolean
05-20 12:26:27.172 20665 20695 I godot   : METHOD: getBool sig: (Ljava/lang/String;)Z
05-20 12:26:27.175 20665 20695 I godot   : Singl:  SQLBridge Method: getFloat RetVal: float
05-20 12:26:27.175 20665 20695 I godot   : METHOD: getFloat sig: (Ljava/lang/String;)F
05-20 12:26:27.176 20665 20695 I godot   : Singl:  SQLBridge Method: getInt RetVal: int
05-20 12:26:27.176 20665 20695 I godot   : METHOD: getInt sig: (Ljava/lang/String;)I
05-20 12:26:27.178 20665 20695 I godot   : Singl:  SQLBridge Method: getValue RetVal: java.lang.String
05-20 12:26:27.178 20665 20695 I godot   : METHOD: getValue sig: (Ljava/lang/String;)Ljava/lang/String;
05-20 12:26:27.181 20665 20695 I godot   : Singl:  SQLBridge Method: setBool RetVal: void
05-20 12:26:27.181 20665 20695 I godot   : METHOD: setBool sig: (Ljava/lang/String;Z)V
05-20 12:26:27.181 20665 20695 I godot   : Singl:  SQLBridge Method: setFloat RetVal: void
05-20 12:26:27.181 20665 20695 I godot   : METHOD: setFloat sig: (Ljava/lang/String;F)V
05-20 12:26:27.181 20665 20695 I godot   : Singl:  SQLBridge Method: setInt RetVal: void
05-20 12:26:27.181 20665 20695 I godot   : METHOD: setInt sig: (Ljava/lang/String;I)V
05-20 12:26:27.182 20665 20695 I godot   : Singl:  SQLBridge Method: setValue RetVal: void
05-20 12:26:27.182 20665 20695 I godot   : METHOD: setValue sig: (Ljava/lang/String;Ljava/lang/String;)V
05-20 12:26:27.182 20665 20695 D godot   : SQLBridge:Initialized
05-20 12:26:27.182 20665 20695 I godot   : ****^*^*?^*^*class instance 100051
05-20 12:26:27.182 20665 20695 I godot   : LOADING MODULE: org/godotengine/godot/FireBase
05-20 12:26:27.182 20665 20695 I godot   : ****^*^*?^*^*class data 100059
05-20 12:26:27.194 20665 20695 I godot   : Singl:  FireBase Method: alert RetVal: void
05-20 12:26:27.194 20665 20695 I godot   : METHOD: alert sig: (Ljava/lang/String;)V
05-20 12:26:27.204 20665 20695 I godot   : Singl:  FireBase Method: authConfig RetVal: void
05-20 12:26:27.204 20665 20695 I godot   : METHOD: authConfig sig: (Ljava/lang/String;)V
05-20 12:26:27.205 20665 20695 I godot   : Singl:  FireBase Method: content_select RetVal: void
05-20 12:26:27.205 20665 20695 I godot   : METHOD: content_select sig: (Ljava/lang/String;Ljava/lang/String;)V
05-20 12:26:27.207 20665 20695 I godot   : Singl:  FireBase Method: download RetVal: void
05-20 12:26:27.207 20665 20695 I godot   : METHOD: download sig: (Ljava/lang/String;Ljava/lang/String;)V
05-20 12:26:27.208 20665 20695 I godot   : Singl:  FireBase Method: earn_currency RetVal: void
05-20 12:26:27.208 20665 20695 I godot   : METHOD: earn_currency sig: (Ljava/lang/String;I)V
05-20 12:26:27.271 20665 20695 I godot   : Singl:  FireBase Method: upload RetVal: void
05-20 12:26:27.271 20665 20695 I godot   : METHOD: upload sig: (Ljava/lang/String;Ljava/lang/String;)V
05-20 12:26:27.271 20665 20695 I godot   : ****^*^*?^*^*class instance 10005d
05-20 12:26:27.283 20665 20695 I godot   : GL_AMD_compressed_ATC_texture
05-20 12:26:27.283 20665 20695 I godot   : GL_AMD_performance_monitor
05-20 12:26:27.283 20665 20695 I godot   : GL_AMD_program_binary_Z400
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_color_buffer_float
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_color_buffer_half_float
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_debug_label
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_debug_marker
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_discard_framebuffer
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_disjoint_timer_query
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_multisampled_render_to_texture
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_robustness
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_sRGB
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_sRGB_write_control
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_texture_filter_anisotropic
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_texture_format_BGRA8888
05-20 12:26:27.283 20665 20695 I godot   : GL_EXT_texture_sRGB_decode
05-20 12:26:27.284 20665 20695 I godot   : GL_EXT_texture_type_2_10_10_10_REV
05-20 12:26:27.284 20665 20695 I godot   : GL_NV_fence
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_EGL_image
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_EGL_image_external
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_EGL_sync
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_compressed_ETC1_RGB8_texture
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_depth24
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_depth_texture
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_depth_texture_cube_map
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_element_index_uint
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_fbo_render_mipmap
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_fragment_precision_high
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_get_program_binary
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_packed_depth_stencil
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_rgb8_rgba8
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_standard_derivatives
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_texture_3D
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_texture_float
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_texture_half_float
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_texture_half_float_linear
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_texture_npot
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_vertex_array_object
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_vertex_half_float
05-20 12:26:27.284 20665 20695 I godot   : GL_OES_vertex_type_10_10_10_2
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_alpha_test
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_binning_control
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_driver_control
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_extended_get
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_extended_get2
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_perfmon_global_mode
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_tiled_rendering
05-20 12:26:27.284 20665 20695 I godot   : GL_QCOM_writeonly_rendering
05-20 12:26:27.284 20665 20695 I godot   : S3TC: 0 ATITC: 1
05-20 12:26:27.327 20665 20695 I godot   : OpenSL Init OK!
05-20 12:26:27.651 20665 20695 I godot   : Loaded certs from 'res://ca-certificates.crt':  173
05-20 12:26:27.801 20665 20665 D FireBase: Data From File: {"AdMob":false,"Authentication":true,"Invites":false,"RemoteConfig":false,"Notification":false,"Storage":false,"Auth":{"Google":false,"Facebook":true,"FacebookAppID":"1360876217329677"},"Ads":{"BannerAd":false,"BannerGravity":"BOTTOM","BannerAdID":"","InterstitialAd":false,"InterstitialAdID":"","RewardedVideoAd":false,"RewardedVideoAdID":""}}
05-20 12:26:27.801 20665 20665 D FireBase: Firebase Analytics initialized..!
05-20 12:26:27.801 20665 20665 D FireBase: Initializing Firebase Authentication.
05-20 12:26:27.817 20665 20665 D FireBase: Facebook auth initialized.
05-20 12:26:27.817 20665 20665 D FireBase: FireBase initialized.
05-20 12:26:27.819 20665 20665 D FireBase: JSONException, parse error: org.json.JSONException: Value Google of type java.lang.String cannot be converted to JSONObject
05-20 12:26:27.820 20665 20665 D FireBase: Facebook auth initialized.
05-20 12:26:28.504 20665 20695 I godot   : ^_^_^_^_^ resize -1458374368, 1080, 1920
05-20 12:26:30.174 20665 20698 D FireBase: Refreshed token: e51Cbdte06w:APA91bF9uhEq_2uLMaATbmsAIemVoZavGF72wDoBamiCwrM7Nod-kEuYc3PawieRiGC7hDCsxpIf9gtDrh6mHRIDOGYGQe02PS6fMITJPTnX7Vz-MrgnMp3WVJn_P4FZGSMlswku6emz
05-20 12:26:30.174 20665 20698 D FireBase: Token: e51Cbdte06w:APA91bF9uhEq_2uLMaATbmsAIemVoZavGF72wDoBamiCwrM7Nod-kEuYc3PawieRiGC7hDCsxpIf9gtDrh6mHRIDOGYGQe02PS6fMITJPTnX7Vz-MrgnMp3WVJn_P4FZGSMlswku6emz
05-20 12:26:31.323 20665 20665 D FireBase: Auth:SignIn:TAG:4
05-20 12:26:31.323 20665 20665 D FireBase: Auth facebook sign in
05-20 12:26:32.410 20665 20665 D FireBase: onActivityResult: reqCode=64206, resCode=-1
05-20 12:26:32.416 20665 20665 D FireBase: FB:Connected
05-20 12:26:32.416 20665 20665 D FireBase: FB:Handle:AccessToken: {AccessToken token:ACCESS_TOKEN_REMOVED permissions:[public_profile, email]}
05-20 12:26:32.434 20665 20695 I godot   : ^_^_^_^_^ resize -1458374368, 1080, 1920
05-20 12:26:32.480 20665 20665 D FireBase: FB:AccessToken:Changed
05-20 12:26:32.480 20665 20665 D FireBase: FB:AccessToken:Changed
05-20 12:26:32.480 20665 20665 D FireBase: FB:Profile:Changed
05-20 12:26:32.480 20665 20665 D FireBase: FB:Profile:Changed
05-20 12:26:33.751 20665 20665 D FireBase: FB:signInWithCredential:onComplete:true
05-20 12:26:36.941 20665 20695 D FireBase: UserDetails:TAG:4
05-20 12:26:36.941 20665 20695 I godot   : NULL
```

**Comment by RameshRavone** [comment]
Ok, checked it this is not a bug but a delay in update, update your code to

```
func _recive_message(from, key, data):
	if from == "FireBase":
		if key == "GoogleLogin" && data == "true": print("User Signed in.");
		if key == "FacebookLogin" && data == "true":
			print("User Signed in.");
			print(firebase.get_facebook_user())
```
or
```
var signedin = false;
func _recive_message(from, key, data):
	if from == "FireBase":
		if key == "GoogleLogin" && data == "true": print("User Signed in.");
		if key == "FacebookLogin" && data == "true":
			print("User Signed in.");
			signedin = true;

func _on_get_profile_pressed():
	if signedin:
		print(firebase.get_facebook_user())
```

**Comment by alketii** [comment]
Still isn't working.
I think the function _recive_message is never being called,
I've added 
```
func _recive_message(from, key, data):
    print("TEST recive message)
```

and it never gets outputed

**Comment by RameshRavone** [comment]
Fixed with [6aa2eaf](https://github.com/FrogSquare/GodotFireBase/commit/008c93c2c1e288a917830845d44c6b7f5ae68cc8)
